docs/aws/audit/sgaudit/rules/subnet_exists_without_flow_logs_remediation.mdx

```diff
@@ -1,6 +1,272 @@
+        filter=f"resource.type=\"gce_subnetwork\" AND resource.labels.subnetwork_id=\"{subnet_name}\""
+    )
+    
+    response = client.create_sink(parent=parent, sink=log_sink)
+    print(response)
+
+# Example usage
+enable_vpc_flow_logs('your-project-id', 'your-subnet-name')
+```
+
+These scripts will help you ensure that Flow Logs are enabled on your subnets across AWS, Azure, and GCP, thereby preventing the misconfiguration.
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Log in to the AWS Management Console and open the Amazon VPC console at https://console.aws.amazon.com/vpc/.
+
+2. In the navigation pane, choose 'Your VPCs'.
+
+3. Select the VPC for which you want to create a flow log.
+
+4. Choose the 'Flow Logs' tab.
+
+5. If there are no flow logs enabled for the selected subnet, it means that the Flow Logs are not enabled. If there are flow logs present, check the 'Status' column. If the status is 'ACTIVE', it means that the Flow Logs are enabled. If the status is 'ERROR' or 'CREATING', it means that there is a misconfiguration.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure AWS CLI on your local machine. You can do this by following the instructions provided by AWS. Make sure you have the necessary permissions to access the resources.
+
+2. Once the AWS CLI is set up, you can use the following command to list all the VPCs in your account:
+
+   ```
+   aws ec2 describe-vpcs
+   ```
+   This command will return a JSON object with details about all your VPCs.
+
+3. To check if flow logs are enabled on a specific subnet, you need to know the ID of the subnet. You can get this information by running the following command:
+
+   ```
+   aws ec2 describe-subnets
+   ```
+   This command will return a JSON object with details about all your subnets. Look for the "SubnetId" field in the output.
+
+4. Once you have the subnet ID, you can check if flow logs are enabled for that subnet by running the following command:
+
+   ```
+   aws ec2 describe-flow-logs --filter "Name=resource-id,Values=<SubnetId>"
+   ```
+   Replace `<SubnetId>` with the ID of the subnet you are interested in. If flow logs are enabled, this command will return a JSON object with details about the flow logs. If no flow logs are enabled, the command will return an empty JSON object.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install and configure AWS SDK for Python (Boto3):
+   You need to install and configure Boto3 to interact with AWS services. You can install it using pip:
+   ```
+   pip install boto3
+   ```
+   Then, configure your AWS credentials to allow Boto3 to interact with AWS services on your behalf. You can do this by setting the following environment variables:
+   ```
+   AWS_ACCESS_KEY_ID='your_access_key'
+   AWS_SECRET_ACCESS_KEY='your_secret_key'
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>